invisible_cities/detsim/light_tables_c.pyx

```diff
@@ -0,0 +1,171 @@
+import numpy  as np
+import pandas as pd
+
+cimport cython
+cimport numpy as np
+
+from libc.math cimport            sqrt
+from libc.math cimport           floor
+from libc.math cimport round as cround
+
+
+
+from ..         core import system_of_units as units
+from  . light_tables import                  read_lt
+
+cdef class LT:
+    """Base abstract class to be inherited from for all LightTables classes.
+    It needs get_values_ cython method implemented, as well as zbins_ and sensor_ids_ attributes.
+    """
+    cdef double* get_values_(self, const double x, const double y, const int sensor_id):
+        raise NotImplementedError
+
+    @property
+    def zbins(self):
+        return np.asarray(self.zbins_)
+    @property
+    def sensor_ids(self):
+        return np.asarray(self.sensor_ids_)
+
+    def get_values(self, const double x, const double y, const int sns_id):
+        """ This is only for using within python"""
+        cdef double* pointer
+        pointer = self.get_values_(x, y, sns_id)
+        if pointer!=NULL:
+            return np.asarray(<np.double_t[:self.zbins_.shape[0]]> pointer)
+        else:
+            return np.zeros(self.zbins_.shape[0])
+
+
+def get_el_bins(el_pitch, el_gap):
+    return np.arange(el_pitch/2., el_gap, el_pitch).astype(np.double)
+
+cdef class LT_SiPM(LT):
+    cdef readonly:
+        double [:] snsx
+        double [:] snsy
+    cdef:
+        double [:, ::1] values
+        double psf_bin
+        double max_zel
+        double max_psf
+        double max_psf2
+        double inv_bin
+        double active_r2
+
+    def __init__(self, *, fname, sipm_database, el_gap=None, active_r=None):
+        lt_df, config_df, el_gap, active_r = read_lt(fname, 'PSF', el_gap, active_r)
+        lt_df.set_index('dist_xy', inplace=True)
+        self.el_gap    = el_gap
+        self.active_r  = active_r
+        self.active_r2 = active_r**2
+
+        el_pitch  = float(config_df.loc["pitch_z"].value) * units.mm
+        self.zbins_    = get_el_bins(el_pitch, el_gap)
+        self.values    = np.array(lt_df.values/len(self.zbins_), order='C', dtype=np.double)
+        self.psf_bin   = float(lt_df.index[1]-lt_df.index[0])
```

Comment:
Add a comment like "pdf bin size inferred from the difference between the distances of the first two entries".

Should you be using the unit system there for consistency?